mixxx-now-playing.sh

```diff
@@ -1,46 +1,105 @@
 #!/usr/bin/env bash
 
+# ====================================================================
+# ==== CUSTOMIZE AS NEEDED FOR YOUR PREFERRED PLAYER
+
+# Mixxx (default)
+PROCNAME='Mixxx'
+TITLEPATTERN='(.+)| Mixxx'
+
+# VLC
+#PROCNAME='VLC'
+#TITLEPATTERN='(.+) - VLC media player'
+
+# TODO: discover proc names and title patterns for Traktor and VirtualDJ
+
 TXTFILE=~/mixxx-now-playing.txt
 
+# ====================================================================
+# ==== DO NOT MODIFY BELOW HERE
+
 touch $TXTFILE
-# TODO: fail here if the file is not writeable
+
+# fail here if the file is not writeable
+[ -w $TXTFILE ] && echo "Capturing track titles to [$TXTFILE]" || { echo "Cannot write to [$TXTFILE]"; exit; }
 
 echo " "  > $TXTFILE
 OS=`uname`
 
 echo "Detected OS: $OS"
 
-while true; do
-	while pgrep -i mixxx > /dev/null; do
-	
-	if [ $OS == "Linux" ]; then
-		xdotool search --name "\| Mixxx" getwindowname |
-		cut -d\| -f1 |
+quartzListAllWindows=$'import Quartz\nprint(Quartz.CGWindowListCopyWindowInfo(Quartz.kCGWindowListExcludeDesktopElements|Quartz.kCGWindowListOptionAll,Quartz.kCGNullWindowID))'
```

Comment:
better than quartz...db query, the db path will be different for windows/whatever for mac:

sqlite3 ~/Library/Application\ Support/Mixxx/mixxxdb.sqlite "select library.artist, library.title, library.year from library where library.id = (select PlaylistTracks.track_id from PlaylistTracks where id = (select max(id) from PlaylistTracks));"

Reply:
was forced to figure this out as quartz was causing mixxx to freeze for me at random for who knows why my setup is "quirky"